logic/src/jvmTest/kotlin/com/wire/kalium/logic/feature/e2ei/PemCertificateDecoderTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.feature.e2ei
+
+import io.mockative.Mock
+import io.mockative.any
+import io.mockative.classOf
+import io.mockative.given
+import io.mockative.mock
+import io.mockative.once
+import io.mockative.verify
+import org.junit.Test
+import java.security.cert.CertificateException
+import kotlin.test.assertEquals
+
+class PemCertificateDecoderTest {
+
+    @Test
+    fun givenAValidCertificate_whenDecodingIt_thenReturnCertificateObject() {
```

Comment:
question: can you add an actual comparison here on the data you expect? based on the PEM string you passed.